src/Input/Select/Select.test.tsx

```diff
@@ -360,3 +360,196 @@ it('up and down arrow keys can be used to navigate the options menu', () => {
+        </Select>
+        <div>outside</div>
+      </React.Fragment>
+    );
+
+    const { queryByTestId, queryByRole, queryByText } = render(<Component />);
+    const selectInput = queryByRole('combobox') as HTMLInputElement;
+    const selectList = queryByTestId('listbox');
+    const outside = queryByText('outside');
+
+    fireEvent.focus(selectInput);
+    expect(selectList.hasAttribute('open')).toEqual(true);
+    fireEvent.click(outside);
+    expect(selectList.hasAttribute('open')).toEqual(false);
+    expect(selectInput).not.toHaveFocus();
+  });
+
+  it('should remove click outside listener when unmount', () => {
+    const addEventListenerSpy = jest.spyOn(document, 'addEventListener');
+    const removeEventListenerSpy = jest.spyOn(document, 'removeEventListener');
+
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside">
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { unmount } = render(<Component />);
+    unmount();
+    expect(removeEventListenerSpy.mock.calls[0][1]).toEqual(
+      addEventListenerSpy.mock.calls[0][1]
+    );
+    addEventListenerSpy.mockRestore();
+    removeEventListenerSpy.mockRestore();
+  });
+});
+
+describe('<Select/> onBlur onFocus', () => {
+  it('should call onBlur and onFocus', () => {
+    const onBlurSpy = jest.fn();
+    const onFocusSpy = jest.fn();
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside" onBlur={onBlurSpy} onFocus={onFocusSpy}>
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, asFragment } = render(<Component />);
+    const selectInput = queryByRole('combobox') as HTMLInputElement;
+
+    fireEvent.focus(selectInput);
+    expect(onFocusSpy).toHaveBeenCalledTimes(1);
+    // use snapshot to validate isFocus
+    expect(asFragment()).toMatchSnapshot();
+
+    fireEvent.blur(selectInput);
+    expect(onBlurSpy).toHaveBeenCalledTimes(1);
+    // use snapshot to validate floating
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
+
+describe('<Select/> handleValueChange', () => {
+  it('input value should be the same as passed prop value', async () => {
+    const Component = ({ value }: { value: string }) => (
+      <React.Fragment>
+        <Select label="handleValueChange" value={value}>
+          <Select.Option value="handleValueChange">
+            handleValueChange
+          </Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const mockValue = 'mock';
+    const { queryByRole, rerender } = render(<Component value={mockValue} />);
+    const selectInput = queryByRole('combobox');
+    expect(selectInput.getAttribute('value')).toEqual(mockValue);
+    rerender(<Component value="" />);
+    expect(selectInput.getAttribute('value')).toEqual('');
```

Comment:
Could we test the `value` change start from an empty string?  So the change flow might be "" -> "mock" -> "".
Because I have a workaround for the case that the `value` prop doesn't apply because the initial value is an empty string.
https://gitlab.glints.com/glints/glints-dst/-/merge_requests/1485/diffs#78d78e4d84be8f56d70d434e1b9e2859becb3501_0_77